Zend/zend_compile.c

```diff
@@ -1469,10 +1470,22 @@ static uint32_t zend_get_class_fetch_type_ast(zend_ast *name_ast) /* {{{ */
 
 static void zend_ensure_valid_class_fetch_type(uint32_t fetch_type) /* {{{ */
 {
-	if (fetch_type != ZEND_FETCH_CLASS_DEFAULT && !CG(active_class_entry) && zend_is_scope_known()) {
-		zend_error_noreturn(E_COMPILE_ERROR, "Cannot use \"%s\" when no class scope is active",
-			fetch_type == ZEND_FETCH_CLASS_SELF ? "self" :
-			fetch_type == ZEND_FETCH_CLASS_PARENT ? "parent" : "static");
+	if (fetch_type != ZEND_FETCH_CLASS_DEFAULT && zend_is_scope_known()) {
+		if (!CG(active_class_entry)) {
+			zend_error_noreturn(
+				E_COMPILE_ERROR,
+				"Cannot use \"%s\" when no class scope is active",
+				fetch_type == ZEND_FETCH_CLASS_SELF ? "self" :
+				fetch_type == ZEND_FETCH_CLASS_PARENT ? "parent" : "static"
+			);
+		} else if (fetch_type == ZEND_FETCH_CLASS_PARENT
+		           && !CG(active_class_entry_parent_name))
+		{
+			zend_error_noreturn(
+				E_STRICT,
```

Comment:
We no longer introduce E_STRICT warnings, please use E_WARNING instead.

Reply:
E_WARNING does not mean this will get changed in a future release. E_DEPRECATED seems odd. I think E_STRICT is the best choice, and unless my memory is failing me we never voted on not adding new E_STRICT; only to remove existing ones.

Reply:
We decided to remove the E_STRICT category in https://wiki.php.net/rfc/reclassify_e_strict. If you'd like to go against this prior decision, this change is going to need an RFC.

Imho either E_DEPRECATED or E_WARNING are sensible choices.

Reply:
This does not match my memory. I was pretty certain we only voted to reclassify existing ones, not remove the category entirely. I will choose `E_DEPRECATED`.